src/microbe_stage/systems/ProcessSystem.cs

```diff
@@ -294,33 +299,36 @@ public static EnergyBalanceInfo ComputeEnergyBalance(IReadOnlyList<OrganelleTemp
             hexCount += organelle.Definition.HexCount;
         }
 
-        result.BaseMovement = Constants.BASE_MOVEMENT_ATP_COST * hexCount;
+        var baseMovement = Constants.BASE_MOVEMENT_ATP_COST * hexCount;
+        result.BaseMovement += baseMovement;
 
         if (includeMovementCost)
         {
             // Add movement consumption together
-            result.AddConsumption("baseMovement", result.BaseMovement);
-            result.TotalMovement = movementATPConsumption + result.BaseMovement;
+            result.AddConsumption("baseMovement", baseMovement);
+            result.TotalMovement += movementATPConsumption + baseMovement;
         }
         else
         {
             result.TotalMovement = -1;
         }
 
         // Add osmoregulation
-        result.Osmoregulation = Constants.ATP_COST_FOR_OSMOREGULATION * hexCount *
+        var osmoregulation = Constants.ATP_COST_FOR_OSMOREGULATION * hexCount *
             membrane.OsmoregulationFactor;
 
         if (isPlayerSpecies)
         {
-            result.Osmoregulation *= worldSettings.OsmoregulationMultiplier;
+            osmoregulation *= worldSettings.OsmoregulationMultiplier;
```

Comment:
Doesn't this result in the results not combining correctly? So for example if we have two cells that have the balance calculated for the overall one, and it is the player species. The first cell might have 20 osmoregulation which is adjusted to 16. Then we calculate the second cell as 20+16=36 and again apply the osmoregulation we get: 28.8 **but** we should have actually ended up with 32 (2 * 16).

So I think this has to "unadjust" the osmoregulation value before this is added to `result.Osmoregulation`.

Something like:
```
if (isPlayerSpecies && result.Osmoregulation != 0)
    result.Osmoregulation *= 1 / worldSettings.OsmoregulationMultiplier
```
(I may have gotten that last line of math wrong as I didn't test it).

Reply:
Ah, you are right of course I had a thought mixup when reading the earlier line and already thinking it was using the result.osmoregulation. So as written it should work. Sorry about the confusion. I'll mark this discussion as resolved.